### ISSUE TITLE
Feature Request: Temporarily Name Unsaved Tabs

### ISSUE THREAD

**Issue by fhaag** [notepad-plus-plus/notepad-plus-plus#3994]
### Description of the Issue

I frequently use Notepad++ for temporarily viewing various pieces of text. One activity where this comes up is when examining the network communication of web applications I am developing. In that case, I will copy and paste various chunks of JSON into several Notepad++ tabs, taking advantage of syntax highlighting, formatting (via plugins), full-text search, and the fact that if I shutdown the computer and go home, the tabs will still be there the next day when I reopen Notepad++.

While all of this is very convenient, one of the few gripes is that these temporary tabs are labeled *new 1*, *new 2*, and so on. Note that I usually do not want to save these tabs - neither do I want to pick (and navigate to/copy the path to) a location for saving them, nor do I want to clean up these files once I am done with the data. By all means, I intend to leave these tabs unsaved.

### Steps to Reproduce the Issue

1. Open some new tabs without saving.
1. Try to tell which is which at one glance (i.e. without first hovering over the tabs to look at the popups one by one).

### Expected Behavior

One possible solution I could imagine would be to let users set a temporary tab name (a "tab display name"). If set, that name would be displayed instead of the usual tab title - possibly formatted in italic letters or somehow differently, to make it clear at first sight that this is not a file name.

Note that this can also be useful when working with saved files: Every now and then, I have to deal with several files of the same name (in different directories). In that situation, being able to rename the tabs to say "original", "first try", and "new copy", or "Module X", "Module Y", "Module Z", etc. would be very helpful, too.

### Actual Behavior

Currently, each tab automatically shows *new (number)* or the filename of the file.

### Debug Information

Notepad++ v7.5.1   (32-bit)
Build time : Aug 29 2017 - 02:35:41
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 8.1 (64-bit)
Plugins : JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer-win32.dll NppTextFX.dll PluginManager.dll UniversalIndentGUI_NPP.dll XMLTools.dll

**Comment by ph0rn** [comment]
IMO just do what for example Sublime does and have the temporary tab name be the first line of the document. Or at least have this be an option

**Comment by kmurray24** [comment]
I would really, really, like this feature!  This would fill in a big hole in my workflow from day to day!
I think a lot of people just use the tabs for temporary copy of data and being able to reference this the next day without having to save would be a big help!

**Comment by sasumner** [comment]
@kmurray24 

Well, you can currently leave it hanging around from day to day **without having to save**, although it would still have the current "new #" name to it.   Tick the "Backup" option "Enable session snapshot..." and then any "new #" tabs you have will persist from run to run of Notepad++.

**Comment by kmurray24** [comment]
Thanks for the comment.  Only problem is that I am usually comparing similar files and would like to have both tabs named differently for reference.  e.g. "New File" "Old File"  "Source File" "Target File"..  
Thanks!

**Comment by fhaag** [comment]
> IMO just do what for example Sublime does and have the temporary tab name be the first line of the document. Or at least have this be an option

That's not terribly helpful when you're looking at, say, JSON files (where the first line will always be `{`).

**Comment by sasumner** [comment]
@SinghRajenM I see you did PR #5311 for this and it looks like it might make it in.  I have many prewritten scripts that check a filename of "new(space)(number" to avoid doing actions on a tab that doesn't have a real disk file associated with it.  What is the best new criterion to check for for these type of files going forward?  Check for absence of backslash? What do you think?

**Comment by SinghRajenM** [comment]
@sasumner As of now there are two solutions
1. Check for the presence of the file (e.g. [PathFileExists](https://docs.microsoft.com/en-us/windows/desktop/api/shlwapi/nf-shlwapi-pathfileexistsa)). In this case, filename will contains on tab name (e.g. "New 1" or may be "Dollar" now ), so that `PathFileExists` will return false for that. Recommended way.
2. Of-course as you suggested, check for absence of backslash.

One question to you:
> I have many prewritten scripts that check a filename of "new(space)(number"

What if a tab name as "New 1", but saved on "C:\User\xyz\Destop\New 1.txt", how prewritten script works? I feel, those scripts should check for file presence instead of checking some pattern e.g. "new(space)(number"

**Comment by sasumner** [comment]
What I was speaking of specifically was Pythonscript, doing a `notepad.getCurrentFilename().startswith("new ")`. Note the trailing space.  This works well as long as the existing scheme is in place.  So I guess I was a bit wrong when I said "new(space)(number)" (meaning in regex terms `^new\s+\d+`).  I checked my scripts and I don't ever do the latter method, always the former.

I was just musing on how to approach this if the rename tab thing gets adopted...  Probably best to check for file existence as you say.  Thanks for your thoughts.

**Comment by SinghRajenM** [comment]
@donho we are ok to close this issue.

**Comment by SinghRajenM** [comment]
@MetaChuh we are ok to close this issue.

**Comment by MetaChuh** [comment]
@SinghRajenM 
thank you

**Comment by fhaag** [comment]
Hello,

First of all, thank you very much for implementing this feature for unsaved tabs. I really appreciate it and I can confirm the feature is very helpful during many of my text-based activities.

The only small gripe is that UI-wise, the functionality has been integrated via the "Rename" command and therefore cannot readily be extended to also work for saved tabs in the future. But that is indeed not the focus of this feature request, so I agree with the closure.

Thank you & Regards

**Comment by sasumner** [comment]
@fhaag said

>extended to also work for saved tabs

Are you meaning taking a named (in the operating system) file and then moving it to an unnamed state (whether the name is "new 4" or some custom name given by the user)?

**Comment by fhaag** [comment]
@sasumner: Not quite, I mean opening a file as usual, but changing the displayed tab title in Notepad++ without renaming the underlying file.

Imagine opening several [`Web.config` files](https://support.microsoft.com/en-us/help/815179/how-to-create-the-web-config-file-for-an-asp-net-application) in an ASP.NET IIS website. All the Notepad++ tabs will have the same names, and renaming the actual files is out of the question, as the name must remain `Web.config` for the website to work.

**Comment by anthonyleestark** [comment]
> IMO just do what for example Sublime does and have the temporary tab name be the first line of the document. Or at least have this be an option

I like this idea
